docs/getting_started/module.md

```diff
@@ -0,0 +1,259 @@
+# Gosoline modules
+
+We saw in the previous section, [Kernels](kernel.md), that every Gosoline application involves _Kernel_ object running one or more _Modules_ which have been added to it.
+
+![kernel_lifecycle](module_1.svg)
+
+When a _Kernel_ starts running, it will launch a goroutine for each module, and each such goroutine runs the middleware configured for that _Kernel_, along with the module itself.
+
+A Gosoline _Module_ is any structure implementing the  _Module_ interface:
+
+[embedmd]:# (../../pkg/kernel/module.go /type Module interface/ /\n}/)
+```go
+type Module interface {
+	// Run the module. If the provided context is canceled you have a
+	// few seconds (configurable with kernel.killTimeout) until your module
+	// is killed (via exit(1)). If you return from Run, it is assumed that
+	// your module is done executing and (depending on the type of your module)
+	// this might trigger a kernel shutdown. If you return an error, a kernel
+	// shutdown is also triggered.
```

Comment:
That would be the correct behavior, but I think right now only the stage of the module is killed, if there are other modules keeping the kernel alive in other stages, the kernel will continue to run. This is however something we should check and fix, not document